lib/json/JsonGenerator.js

```diff
@@ -164,7 +164,7 @@ class JsonGenerator extends Generator {
 		const jsonStr = stringifySafe(finalJson);
 		const jsonExpr =
 			jsonStr.length > 20 && typeof finalJson === "object"
-				? `JSON.parse(${JSON.stringify(jsonStr)})`
+				? `JSON.parse('${jsonStr.replace("\\", "\\\\").replace("'", "\\'")}')`
```

Comment:
```suggestion
				? `JSON.parse('${jsonStr.replace(/[\\']/g, "\\$&")}')`
```

Reply:
replace() with a string only replaces the first result. You need to use a regexp with g flag. It's also possible to this in a single operation.